main/src/cgeo/geocaching/settings/OAuthPreference.java

```diff
@@ -78,20 +83,53 @@ public boolean onPreferenceClick(final Preference preference) {
 
     }
 
+    private ImmutablePair<Integer, Integer> getKeysIds() {
```

Comment:
Keys ids don't seem to fit into `OAuthParameters` (since there are already keys themselves), there could be some place in `OCPreferenceKeys`, but `OCPreferenceKeys` is OC-specific, so it will require some additional efforts to extract OAuth functionality from OC-only to more abstract level.

I've added this pair of parameters to `OAuthActivityMapping` itself, probably looks a bit better now.